### PR TITLE
Update RAM ms-patches file.

### PR DESCRIPTION
**Pull request by JohnTortugo**
Please, review this Pull Request to `update ms-patches/reduce-allocation-merges.yml` file with details of last changes.

### CODE REVIEW

File: ms-patches/reduce-allocation-merges.yml [hunk 1]
```diff
@@ -11,4 +11,12 @@ title: Reduce Allocation Merges
   - Add a new Class to support the rematerialization of SR objects that are part of a merge;
   - Patch HotSpot to be able to serialize and deserialize debug information related to allocation merges;
   - Patch C2 to generate unique types for SR objects participating in some allocation merges.
+  - Fix: Incorrect rematerialization of scalar replaced objects in C2
```
Comment (swesonga) [inline ms-patches/reduce-allocation-merges.yml:14]: Why doesn't this note mention openjdk/jdk#17562 like the jdk11u PR?
Same question for jdk17u

File: ms-patches/reduce-allocation-merges.yml [hunk 2]
```diff
@@ -11,4 +11,12 @@ title: Reduce Allocation Merges
...
+  - Fix: Segfault during deoptimization of C2-compiled code
+  - Fix: VM crash during heap dump after JDK-8287061
+  - Fix: AllocationMergesTests.java fails with "IRViolationException: There were one or multiple IR rule failures."
```
Comment (swesonga) [inline ms-patches/reduce-allocation-merges.yml:17]: Was this not backported to jdk11?

File: ms-patches/reduce-allocation-merges.yml [hunk 3]
```diff
@@ -11,4 +11,12 @@ title: Reduce Allocation Merges
...
+  - Fix: Live Node limit exceeded limit after JDK-8316991
```
Comment (dhanalla) [inline ms-patches/reduce-allocation-merges.yml:18]: NIT: word limit repeated in
"- Fix: Live Node limit exceeded limit after JDK-8316991"

**Review by dhanalla — APPROVED** [review]

File: ms-patches/reduce-allocation-merges.yml (hunk 2 above)
Comment (JohnTortugo) [reply ms-patches/reduce-allocation-merges.yml:17]: IR-base test framework not available in jdk11.

File: ms-patches/reduce-allocation-merges.yml (hunk 1 above)
Comment (JohnTortugo) [reply ms-patches/reduce-allocation-merges.yml:14]: That was a "typo" in JDK11 PR. I just copied the titles of the PRs with the fixes and it ended copying that part as well.

**Pull request merged**